pkg/controller/jobs/job/job_controller.go

```diff
@@ -230,6 +246,7 @@ func (j *Job) RunWithPodSetsInfo(podSetsInfo []podset.PodSetInfo) error {
 			j.Spec.Completions = j.Spec.Parallelism
 		}
 	}
+	addPreservedLabelsToInfo(j.Spec.Template.Labels, &info)
```

Comment:
Can you elaborate why this is needed? Some particular questions:
1. Why just this subset of whitelist labels is copied?
2. Are there any specific labels we don't want to copy
3. Why don't we use of blacklist?

Reply:
Comment extended.